src/generators/clientContextFileGenerator.ts

```diff
@@ -174,16 +174,33 @@ const writeStatements = (lines: string[], shouldAddBlankLine = false) => (
 };
 
 function writeDefaultOptions(hasCredentials: boolean, hasLRO: boolean) {
-  const policies = ` 
-  const defaultPipelines = Array.isArray(options.requestPolicyFactories)
-  ? options.requestPolicyFactories
-  : coreHttp.createPipelineFromOptions(options)
-      .requestPolicyFactories as coreHttp.RequestPolicyFactory[];
-
-  options = {
-      ...options,
-      requestPolicyFactories: [lroPolicy(), ...defaultPipelines]
-    };`;
+  const addLROPolicy = hasLRO
+    ? `
+    // Building the rquest policy fatories based on the passed factories and the
+    // any required factories needed by the client.
+    if (Array.isArray(options.requestPolicyFactories)) {
+      // When an array of factories is passed in, we'll just add the required factories,
+      // in this case lroPolicy(). It is important to note that passing an array of factories
+      // to a new client, bypasses core-http default factories. Just the pipelines provided will be run.
```

Comment:
What is the passed in values already have an lroPolicy?

Reply:
Yeah, I think this is one of the drawbacks of the current pipelines design, that we cannot identify what pipelines are already in place. We'll be able to handle this once we move to core-client and core-https.

Meanwhile we don't have a way to make sure we don't duplicate the lroPolicy. In the bright side, the lroPolicy is not publicly exposed so having someone passing it in when creating a new client wouldn't be too common. I think we should be safe in the meantime while we get migrated to corev2